pkg/pillar/cmd/zedrouter/server.go

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// A http server providing meta-data information to application instances
+// at http://169.254.169.254. The source IP address is used to tell
+// which app instance is sending the request
+
+package zedrouter
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/lf-edge/eve/pkg/pillar/iptables"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+// Provides a json file
+type networkHandler struct {
+	ctx *zedrouterContext
+}
+
+// Provides a LF-terminated text
+type externalIPHandler struct {
+	ctx *zedrouterContext
+}
+
+// Provides a LF-terminated text
+type hostnameHandler struct {
+	ctx *zedrouterContext
+}
+
+func createServer4(ctx *zedrouterContext, bridgeIP string, bridgeName string) error {
+	mux := http.NewServeMux()
+	nh := &networkHandler{ctx: ctx}
+	mux.Handle("/eve/v1/network.json", nh)
+	ipHandler := &externalIPHandler{ctx: ctx}
+	mux.Handle("/eve/v1/external_ipv4", ipHandler)
+	hostnameHandler := &hostnameHandler{ctx: ctx}
+	mux.Handle("/eve/v1/hostname", hostnameHandler)
+
+	targetPort := 80
+	subnetStr := "169.254.169.254/32"
+	target := fmt.Sprintf("%s:%d", bridgeIP, targetPort)
+	log.Noticef("add NAT to target %s", target)
+	if err := iptables.IptableCmd(log, "-t", "nat", "-I", "PREROUTING",
+		"-i", bridgeName, "-p", "tcp", "-d", subnetStr,
+		"--dport", strconv.Itoa(targetPort),
+		"-j", "DNAT", "--to-destination", target); err != nil {
+		log.Errorf("failed to add DNAT: %s", err)
+	}
+	if err := iptables.IptableCmd(log, "-t", "nat", "-I", "POSTROUTING",
```

Comment:
Wouldn't the NAT session take care of reverse translation in this case?

Reply:
That is a good question. Looking at the counters it doesn't seem like we need the SNAT (I've assumes that SNAT and DNAT are stateless but that doesn't seem to be the case; not that we have SNAT+DNAT pair of rules elsewhere.)
Should I fix it just for this one?

Reply:
Fixed.